File: Heat/ModelBinders/DecimalModelBinder.vb
Imports System.Globalization

Public Class DecimalModelBinder
    Implements IModelBinder


    Public Function BindModel(controllerContext As ControllerContext, bindingContext As ModelBindingContext) As Object Implements IModelBinder.BindModel
        Dim valueResult As ValueProviderResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName)
        Dim modelState As ModelState = New ModelState()
        modelState.Value = valueResult

        Dim actualValue As Object = Nothing

        Try
            actualValue = Convert.ToDecimal(valueResult.AttemptedValue, CultureInfo.CurrentCulture)

        Catch ex As FormatException
            modelState.Errors.Add(ex.Message)

        End Try
        bindingContext.ModelState.Add(bindingContext.ModelName, modelState)
        Return actualValue

    End Function
End Class

File: Heat.ConvertedToC#/ModelBinders/DecimalModelBinder.cs
using System;
using System.Web.Mvc;
using System.Globalization;
namespace Heat
{

    public class DecimalModelBinder : IModelBinder
	{


		public object BindModel(ControllerContext controllerContext, ModelBindingContext bindingContext)
		{
			ValueProviderResult valueResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName);
			ModelState modelState = new ModelState();
			modelState.Value = valueResult;

			object actualValue = null;


			try {
				actualValue = Convert.ToDecimal(valueResult.AttemptedValue, CultureInfo.CurrentCulture);

			} catch (FormatException ex) {
				modelState.Errors.Add(ex.Message);

			}
			bindingContext.ModelState.Add(bindingContext.ModelName, modelState);
			return actualValue;

		}
	}
}
